### PR TITLE
 Sign up component

### PR DESCRIPTION
**Pull request by farah2003**
- [x] create Form and its component
- [x]  style signup page
- [x]  create store and userReducer and set user information in store
- [x] fetch user data to backend
- [x]  create error message
image

### CODE REVIEW

File: client/src/Component/signUp/index.jsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+// import { useNavigate } from 'react-router-dom';
+import axios from 'axios';
+import {
+  Form, Button, Input, message, Space, Typography,
+} from 'antd';
```
Comment (Raghad-Mezied) [inline client/src/Component/signUp/index.jsx:7]: add each one in a new line

File: client/src/Component/signUp/index.jsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+import { signUp } from '../../feature/userReducer';
+import './style.less';
+
+function SignUp() {
```
Comment (Raghad-Mezied) [inline client/src/Component/signUp/index.jsx:11]: Convert this to arrow function with function declaration

File: client/src/Component/signUp/index.jsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  // const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const [userInfo, setUserInfo] = useState({
+    name: '', email: '', password: '', phone: '',
+  });
+
+  const Registor = async () => {
+    try {
+      const response = await axios.post('/api/signup', userInfo);
+      const data = response.data.data[0];// user info object
```
Comment (Raghad-Mezied) [inline client/src/Component/signUp/index.jsx:21]: Destructure the data here.

File: client/src/feature/userReducer.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      name: '', id: '', isAdmin: false,
+    },
+  },
+  reducers: {
+    signUp: (state, action) => {
+      state.userInfo = action.payload;
+    },
+  },
+
+});
```
Comment (Raghad-Mezied) [inline client/src/feature/userReducer.js:16]: Create a folder called redux, inside it put the app and the feature, inside the feature create another folder called user and put this file inside it, but rename it to userSlice.

File: client/src/Component/signUp/index.jsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+                  required: true,
+                  message: 'Please confirm your password!',
+                },
+                ({ getFieldValue }) => ({
+                  validator(_, value) {
+                    if (!value || getFieldValue('password') === value) {
+                      return Promise.resolve();
+                    }
+
+                    return Promise.reject(new Error('The two passwords that you entered do not match!'));
```
Comment (Raghad-Mezied) [inline client/src/Component/signUp/index.jsx:152]: Move this to separated function and then use it here

File: client/src/Component/signUp/index.jsx [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+      </div>
+      <div className="form-conatainer-signup">
+        <Space
+          direction="vertical"
+          style={{
+            height: '100%',
+            justifyContent: 'center',
+            width: '100%',
+            alignItems: 'center',
+          }}
```
Comment (Raghad-Mezied) [inline client/src/Component/signUp/index.jsx:52]: Move all the style to the css file.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: client/src/Component/signUp/index.jsx [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+                {
+                  required: true,
+                  message: 'Please input your nickname!',
+                  whitespace: true,
+                },
+              ]}
+            >
+              <Input
+                placeholder="Name"
+                onChange={(e) => setUserInfo({ ...userInfo, name: e.target.value })}
```
Comment (Raghad-Mezied) [inline client/src/Component/signUp/index.jsx:77]: move this to a separated function and then use it

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: client/src/Component/signUp/index.jsx [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+              dependencies={['password']}
+              rules={[
+                {
+                  required: true,
+                  message: 'Please confirm your password!',
+                },
+                ({ getFieldValue }) => confirmPasswordValidation(getFieldValue),
+              ]}
+            >
+              <Input.Password placeholder="Confirm password" />
```
Comment (Raghad-Mezied) [inline client/src/Component/signUp/index.jsx:160]: Destructure the password, item and Tilte

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: client/.eslintrc.json [hunk 9]
```diff
@@ -24,6 +24,8 @@
         "no-param-reassign": [
             "error",
             { "props": true, "ignorePropertyModificationsForRegex": ["^draft"] }
-          ]
+          ],
+          "no-param-reassign": 0
+          
     }
 }
```
Comment (Raghad-Mezied) [inline client/.eslintrc.json:31]: Add this rule for function decleration.
```js
        "react/function-component-definition": {
            "namedComponents": "function-declaration"
        }
```
And why there is two values for  `no-param-reassign`?

File: client/src/Component/signUp/index.jsx [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+  });
+
+  const handleChange = ({ target: { name, value } }) => {
+    setUserInfo({ ...userInfo, [name]: value });
+  };
+  return (
+    <div className="sign-up-container">
+      <div className="img-side-sgin-up">
+        <Typography.Title
+          style={{ color: '#fff' }}
```
Comment (Raghad-Mezied) [inline client/src/Component/signUp/index.jsx:61]: Move the style to style files

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Mohammed-Hamada — APPROVED** [review]

**Review by Mohammed-Hamada — APPROVED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Review by DeenaAlghazali — APPROVED** [review]

**Review by HaniOlwan — APPROVED** [review]
Nice work

**Pull request merged**